--- Makefile	2017-04-25 13:16:43.000408158 -0500
+++ Makefile-zc	2017-04-25 14:03:19.653159599 -0500
@@ -1,12 +1,20 @@
 include ../Config
 
+ZCPATH = ../../Z-checker/zc-install
+ZCFLAG  = -I$(ZCPATH)/include $(ZCPATH)/lib/libzc.a
+
 TARGET = ../bin/zfp
+TARGET2 = ../bin/zfp-zc
 
-all: $(TARGET)
+all: $(TARGET) $(TARGET2)
 
 $(TARGET): zfp.c ../lib/libzfp.a
 	mkdir -p ../bin
-	$(CC) $(CFLAGS) zfp.c -L../lib -lzfp -lm -o $(TARGET)
+	$(CC) $(CFLAGS) zfp.c -L../lib -lzfp -lm -o $(TARGET) 
+
+$(TARGET2): zfp-zc.c ../lib/libzfp.a
+	mkdir -p ../bin
+	$(CC) $(CFLAGS) zfp-zc.c -L../lib -lzfp -lm -o $(TARGET2) $(ZCFLAG)
 
 clean:
-	rm -f $(TARGET) fields.o
+	rm -f $(TARGET) $(TARGET2) fields.o
